wdl/transforms/new-base/src/main/scala/wdl/transforms/base/linking/expression/types/EngineFunctionEvaluators.scala

```diff
@@ -192,6 +193,7 @@ object EngineFunctionEvaluators {
                              (implicit expressionTypeEvaluator: TypeEvaluator[ExpressionElement]): ErrorOr[WomType] = {
       a.param.evaluateType(linkedValues).flatMap {
         case WomArrayType(WomOptionalType(inner)) => WomArrayType(inner).validNel
+        case alreadyNonOptional: WomArrayType => alreadyNonOptional.validNel
```

Comment:
Why is it `case WomArrayType(alreadyNonOptional)` above, but `case alreadyNonOptional: WomArrayType` here?

Reply:
Basically because the result type of `select_first(Array[X?])` is reduced down to `X` whereas the type of `select_all(Array[X?])` is still `Array[X]`